### PR TITLE
[ADD] Website: creating building blocks documentation

### PR DESCRIPTION
**Pull request by Donapi**
taskid-3693628

### CODE REVIEW

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+   .. image:: building_blocks/phone-icon.png
+      :alt: Click the phone icon to view how your website renders on mobile.
+
+Saving a custom building block
+==============================
+
+You can save your favorite custom building blocks for future use on other site pages. Select the
+custom building block you want to save, navigate to the :guilabel:`Customize` tab, and click the
+floppy disk icon to store it for later application.
```
Comment (chiaraprattico) [inline content/applications/websites/website/web_design/building_blocks.rst:120]: floppy disk icon to store it.

**Review by chiaraprattico — CHANGES_REQUESTED** [review]

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+You can replicate a building block by clicking on the duplicate icon. Once duplicated, the new block
+appears on your website beneath the original one.
+
+.. image:: building_blocks/duplicate-container.png
+   :alt: Replicating a building block
+
+Re-ordering building blocks
+===========================
+
+To re-order the building blocks on your site, select the block you want to move and click the up
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:86]:
```suggestion
To reorder building blocks, select the block you want to move and click the up
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+arrow to move it before the previous section or click the down arrow to move it after the following
+section.
+
+You can also use the drag-and-drop icon to move your block manually.
+
+.. image:: building_blocks/re-ordering-blocks.png
+   :alt: Re-order building blocks by using the up and down arrow icons.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:93]:
```suggestion
   :alt: Reorder building blocks by using the up and down arrow icons.
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+Show/Hide on mobile
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:95]:
```suggestion
Visibility of building blocks per device type or condition
```
I would change the heading to speak about all visibility options.

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+===================
+
+You can hide specific elements of your website for mobile devices. To do so, select the element
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:98]:
```suggestion
You can hide building blocks for users on mobile devices or desktops. To do so, select the element
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+to hide, and in the :guilabel:`Customize` tab, scroll down to :guilabel:`Visibility`, and click the
+:guilabel:`Show/Hide on Mobile` icons.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:100]:
```suggestion
:guilabel:`Show/Hide on Mobile` or the :guilabel:`Show/Hide on Desktop` icon.
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+.. image:: building_blocks/show-hide-on-mobile.png
+   :alt: Click the "show/hide on mobile" icons to show or hide some elements on mobile.
+
+You can decide to show/hide with :guilabel:`No condition` or
+:guilabel:`Conditionally`, depending on :guilabel:`Country` (based on Geo IP), :guilabel:`Languages`
+(for multi-language websites), :guilabel:`UTM Campaign`, :guilabel:`UTM Medium`,
+:guilabel:`UTM Source`, or :guilabel:`Users`.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:108]:
```suggestion
You can also show or hide building blocks depending on specific conditions.
To do so, click :guilabel:`No condition` and select :guilabel:`Conditionally` instead. Next, under one of the available,  conditions, select if the building block should be :guilabel:`Visible for` or :guilabel:`Hidden for` specific records.
```
The available conditions depend on your DB, not everyone uses multiple languages or UTM campaigns. I think describing the general process is more interesting, especially if you don't describe what are the options.

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+The elements hidden on mobile/desktop view appear in the :guilabel:`Invisible Elements` section at
+the bottom of the configurator. Click on a specific element to make it re-appear on screen.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:111]:
```suggestion
Some elements can become hidden from your current view. To make a building block visible again, go to the :guilabel:`Invisible Elements` section at the bottom of the configurator and select a building block.
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+.. tip::
+   Click the phone icon at the top of the configurator to see how your website looks on mobile.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:114]:
```suggestion
   Click the phone icon at the top of the configurator to preview how your website would look on a mobile device.
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+   .. image:: building_blocks/phone-icon.png
+      :alt: Click the phone icon to view how your website renders on mobile.
+
+Saving a custom building block
+==============================
+
+You can save your favorite custom building blocks on other site pages for future use. Select the
+custom building block you want to save, navigate to the :guilabel:`Customize` tab, and click the
+floppy disk icon to store it to use it later.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:124]:
```suggestion
You can create custom building blocks that you can reuse elsewhere. To do so, select
the building block you want to save, navigate to the :guilabel:`Customize` tab, and click the
floppy disk icon to save it.
```

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+.. image:: building_blocks/saving-custom-block.png
+   :alt: Saving a custom block to use it elsewhere
+
+Your custom building blocks are available in the :guilabel:`Custom` section of the
+:guilabel:`Blocks` tab. Click on the pen icon to edit their name and easily find them again.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:130]:
```suggestion
Your custom building blocks are available in the :guilabel:`Custom` section of the
:guilabel:`Blocks` tab. You can click the pen icon to edit their name.
```

**Review by xpl-odoo — CHANGES_REQUESTED** [review]
Good job Dona, interesting doc! I didn't change the structure, but I (of course) have plenty of suggestions to make it even better (I hope :D)

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 12]
```diff
@@ -0,0 +1,133 @@
+===============
+Building blocks
+===============
+
+It is possible to add and design your content without coding by dragging and dropping building
+blocks from the website builder onto your web pages. Odoo classifies them into four categories:
+:guilabel:`Structure`, :guilabel:`Features`, :guilabel:`Dynamic Content`, and :guilabel:`Inner
+Content`.
```
[the remark this answers is not in the document]
Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:8]: One page is being created per building block category and the link will be added for each page :)

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+.. note::
+   In the mobile view, only one element per column is used by default for better visibility.
+   However, you can adjust this setting without impacting the desktop view.
+
+Duplicating a building block
+============================
+
+You can replicate a building block by clicking on the duplicate icon. Once duplicated, the new block
+appears on your website beneath the original one.
```
[the remark this answers is not in the document]
Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:78]: I prefer to keep it, as it clarifies how it works.

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+
+It is possible to add and design your content without coding by dragging and dropping building
+blocks from the website builder onto your web pages. Odoo classifies them into four categories:
+:guilabel:`Structure`, :guilabel:`Features`, :guilabel:`Dynamic Content`, and :guilabel:`Inner
+Content`.
+
+.. seealso::
+   - `Tutorial: Design your first webpage <https://www.odoo.com/slides/slide/design-your-first-web-page-1667?fullscreen=1>`_
+   - `Odoo Website <https://www.odoo.com/app/website>`_
+   - :doc:`Website themes <themes>`
```
[the remark this answers is not in the document]
Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:13]: I would specify the Website themes link is part of the documentation. 
--> Should we discuss it with everyone to agree on a single way of doing things?

File: content/applications/websites/website/web_design/building_blocks.rst (hunk 4 above)
Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:95]: I do not really like the title you suggested; it's too long for me. If you can live with it, I would prefer to keep the title I wrote.

Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:95]: or maybe to speak about "visibility on mobile"?

Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:95]: updated as discussed live

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+
+Conditional visibility
+----------------------
+
+You can also conditionally render visible or hidden for certain building blocks. To do so, select
+the element you want to show/hide, go to :guilabel:`Visibility`, and click :guilabel:`Conditionally`.
+Then, choose the conditions to apply based on the :guilabel:`Country` (based on Geo IP),
+:guilabel:`Languages` (for multi-language websites), :guilabel:`UTM Campaign`,
+:guilabel:`UTM Medium`, :guilabel:`UTM Source`, or :guilabel:`Users`. The options available depend
+on your database's settings.
```
Comment (xpl-odoo) [inline content/applications/websites/website/web_design/building_blocks.rst:118]:
```suggestion
You can also hide or show building blocks using other conditions. To do so, select
an element, go to :guilabel:`Visibility`, click :guilabel:`No condition`, and select :guilabel:`Conditionally` instead.
Then, configure the condition(s) to apply by selecting :guilabel:`Visible for` or :guilabel:`Hidden for` and which  :guilabel:`Records` will be impacted.
```
- The first sentence isn't clear.
- By default, you don't see ***Conditionally*** but ***No condition***
- You don't choose conditions based on _xxx_ (a condition). For a condition, you have to choose: which records the condition applies to and if the block should be shown or hidden.

**Review by xpl-odoo — COMMENTED** [review]
We are almost there :D

File: content/applications/websites/website/web_design/building_blocks.rst [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+You can also use the drag-and-drop icon to move your block manually.
+
+.. image:: building_blocks/re-ordering-blocks.png
+   :alt: Reorder building blocks by using the up and down arrow icons.
+
+Visibility
+==========
+
+Visibility on mobile
```
[the remark this answers is not in the document]
Comment (Donapi) [reply content/applications/websites/website/web_design/building_blocks.rst:93]: updated as discussed live

**Comment by xpl-odoo** [comment]
@robodoo r+

**Pull request closed without merging**